### ISSUE TITLE
problems with macOS Monterey

### ISSUE THREAD

**Issue by ozster123** [airsdk/Adobe-Runtime-Support#1555]
Having issues with the latest available runtime (AIR runtime - version 33.1.1.533) on macOS Monterey 12.1 using our AIR application.

- Typing the character "e" in an input field causes the character viewer dialogue to appear. This functionality is normally fn+e.
- The browse for open dialogue only works the first time. Any subsequent invoke locks the application.

I think some other issues have already been resolved. Will there be a runtime release when the issues have been resolved?

**Comment by ajwfrost** [comment]
Yes we've got a few things on Monterey that are causing problems, had hoped to get a fix in for a critical Android bug at the same time but that is taking a bit longer. So we will push ahead with an update to fix these macOS problems and get this uploaded as a shared AIR runtime as well...

Should be available at the start of next week I would expect.

thanks

**Comment by leossmith** [comment]
Hi @ajwfrost I am not sure if this is the same issue. I am trying to compile on a mac mini m1 with monterey and get this error:

![Screenshot 2022-01-13 at 3 12 04 PM](https://user-images.githubusercontent.com/4300876/149336531-109733ed-e259-4ccb-b13b-6d88d5909dbf.png)

**Comment by ajwfrost** [comment]
@leossmith that looks like something new: presumably you're trying to build for the simulator target here...? are you able to confirm what settings it's using (are you packaging via Moonshine? or another IDE?) - or can you set the lib/adt.cfg file to have "DebugOut=true" and then check what the full linker command is from the ~/adt.log file?

thanks

**Comment by ajwfrost** [comment]
@leossmith so I think the issue is here that your linker command is including a request `-ios_version_min 9.0` which is why the linker then says `building for iOS/iOS Simulator`...

This value is likely being provided by one of the ANEs in their linker options. So it may be that it's possible to edit the ANE platform options and remove this manually ... or, we made a change in the latest release (33.1.1.731) relating to this, under #1497, so now if any ANEs (or info additions) specify a minimum iOS version, they will automatically be rolled up into one maximum value and this is then used at the end of the linker command - and for a simulator build, it would then be something like you have there: `-platform_version ios-simulator 12.2 15.0`

So - please check with the latest SDK release..
thanks

**Comment by pmichel-J2S** [comment]
Can reproduce the file open dialog problem 100% since we upgraded to Monterey with the latest Harman SDK (33.1.1.743).

**Comment by ajwfrost** [comment]
@pmichel-J2S are you able to set up your adt.cfg file with the `DebugOut=true` setting and then check what the full linker command is (in `~/adt.log`) please? I wouldn't have expected there to be any `ios_version_min` values being used from 33.1.1.743...

thanks

**Comment by ajwfrost** [comment]
Hold on, sorry @pmichel-J2S  I just read that again ... "reproduce the file open dialog problem" ...? With 33.1.1.743 we should have  all the fixes in there for Monterey (well, 12.1 at least I think it was) - are you able to confirm with a SWF that displays the `Capabilities.version` number and let us know the exact Monterey version number? And if you can clarify what behaviour you're seeing it would be good, there were a few different issues around file dialogs...

thanks

**Comment by pmichel-J2S** [comment]
Thanks for the answer man!

Soooo I feel silly... I just double-checked after you mentioned it was fixed, and it seems my toolchain still had a symlink pointing to an older SDK... just re-checked with the newest and the bug is fixed ! Works like a charm, thanks a lot, and sorry for wasting your time. 😅

**Comment by ajwfrost** [comment]
:-) no worries, good to know. Management of different SDK installations can be a pain ...!

**Comment by rogerclarkmelbourne** [comment]
This bug is still showing as open. Does that mean the bug has yet to be fixed or is there a workaround ?

We're getting reports that our Mac App has problems with the 'D' and 'E' characters being entered, (possibly also 'd' and 'e')

**Comment by ajwfrost** [comment]
The behaviour of the dictionary/emoji dialogs appearing when you're typing a 'd' or 'e' character should be fixed in the 33.1.1.743 releases.. if you're still seeing this, can you please double-check the runtime version being used?

We've got a "to do" still which is to add the function key as a key code so that you can actually get/use these shortcuts (currently we're just swallowing/blocking actual "Fn+x" key combinations -- the original issue was that we didn't know about the 'Fn' accelerator so when a menu had "Fn+D" as a shortcut code, we treated it as if it was just "D" ...)

**Comment by rogerclarkmelbourne** [comment]
Thanks. I was hoping for a work-around that did not involve republishing again, becasue the QA is non-trivial for us every time there is a change to the SDK, and also getting users to install the update is also non-trivial.

**Comment by ajwfrost** [comment]
If you're bundling the runtime with your app then the fix was in there from 33.1.1.633, but if folk are using the shared runtime then it's 33.1.1.743 as we don't issue a new shared runtime every time the SDK updates. Sorry, a bit confusing with the disparity in version numbers.

**Comment by rogerclarkmelbourne** [comment]
In case anyone needs to do this. I have been sent a workaround for both the 'e' and 'd' keys

The 'e' key assignemnt is as suggested by someone in another issue in this repo (sorry I can't remember their name/handle)

The 'd' key can be fixed by assigning Command D to "Start Dictation..."


This information may have originally come from here

https://discussions.apple.com/thread/253297084?page=2

I've also been told that if the 'f' key does not work (in my case it does work), That Command F needs to be assigned to "Enter Full Screen"

But since the problem does not exist on 'f' I've not bothered re-assigning that shortcut

<img width="780" alt="Screen Shot 2022-02-15 at 2 00 01 pm" src="https://user-images.githubusercontent.com/6146984/153985017-67d6efc4-d5f1-4d2a-ba9f-7ec465c5c122.png">

**Comment by aarontheman** [comment]
The workarounds are helpful; thanks -- but Adobe can you please update the app to work properly?

**Comment by aarontheman** [comment]
I am on Air 32.0.0.125. Can anyone point me to a newer Air install for Mac? 
This link appears broken: http://get.adobe.com/air

**Comment by ajwfrost** [comment]
If you use https it takes you to their "state of AIR" page, not sure why it doesn't automatically do that from an http request..

But for the latest AIR runtimes, please use https://airsdk.harman.com/runtime

thanks

**Comment by aarontheman** [comment]
@ajwfrost , thank you so much! Typing issues fixed with latest version on my Mac Mini M1.
